src/main/java/seedu/eylah/expensesplitter/model/person/Amount.java

```diff
@@ -0,0 +1,67 @@
+package seedu.eylah.expensesplitter.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.eylah.commons.util.AppUtil.checkArgument;
+
+import java.math.BigDecimal;
+
+/**
+ * Represents a Person's amount in Expense Splitter in EYLAH.
+ * Guarantees: immutable; is valid as declared in {@link #isValidAmount(BigDecimal)}
+ */
+public class Amount {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Amount should only contains numerical characters and decimal separator.";
+
+    // @@author Gary-reused
+    // Solution below adapted from https://stackoverflow.com/questions/16242449
```

Comment:
Nice idea for using BigDecimal to represent Currency! i understand that using Double may not be as good as it will lead to rounding errors!

Reply:
thanks